soh/soh/GameMenuBar.cpp

```diff
@@ -984,6 +984,20 @@ namespace GameMenuBar {
 
                     UIWidgets::Spacer(0);
 
+                    if (ImGui::BeginMenu("Shooting Gallery")) {
+                        UIWidgets::EnhancementCheckbox("Instant Win", "gInstantShootingGalleryWin");
+                        UIWidgets::Tooltip("Skips the shooting gallery minigame");
+                        UIWidgets::EnhancementCheckbox("No Rupee Randomization", "gConstantAdultGallery");
+                        UIWidgets::Tooltip("Forces the rupee order to not be randomized as adult, making it the same as chlid");
+                        UIWidgets::PaddedEnhancementSliderInt("Child Starting Ammunition: %d", "##cShootingGalleryAmmunition", "gChildShootingGalleryAmmunition", 10, 30, "", 15, false, true, false);
```

Comment:
I like the idea of all the shooting gallery difficulty options being wrapped in a cvar (something like `gCustomizeShootingGallery` works)

The checkbox could show/hide the difficulty options menu, and when unchecked everything goes back to authentic behavior without resetting any player set difficulty options

Reply:
Sounds good. I actually thought about this issue before but finally decided to copy the style of the "Fishing" menu. The "Potion Values" menu already has a similar locks for every single potion but it is also very cluttered.
The "Fishing" menu should probably get a similar switch then, too.

Reply:
I have now added a checkbox to turn on and off all modifications to the shooting gallery behavior. First I wanted to disable the other UI-Elements if the box is not checked, however the sliders seem to have currently no disable-switch. Instead I decided to just remove the other elements if the box is not activated.

Reply:
@briaguya-ai @dcvz How do you like my current solution?

Reply:
did you update it to do disabled instead of just hiding the options when it's disabled? i feel like i might like hiding them better. @dcvz thoughts?

Reply:
Correct. If the checkbox to change the behavior is unchecked it now disables instead of hides the other options. Otherwise the menu would be quite empty if it's turned off (one could move the checkbox out of the shooting gallery submenu, but I don't quite like that solution either since it clutters the difficulty options menu) and the user couldn't know what settings are offered without turning the changed behavior on. That's why I personally think this is the best solution.
If you maintain that hiding them is better I can of course change it again without any issues.

Reply:
in a submenu disabled works, any chance you could add screenshots of the menu to this PR real quick?

Reply:
![shooting_gallery_difficulty_menu2](https://user-images.githubusercontent.com/595497/202032068-9a8b26f8-cde0-4593-a5e6-e95eff1dcac0.png)
![shooting_gallery_difficulty_menu1](https://user-images.githubusercontent.com/595497/202032072-2ab55682-8548-486f-b3a9-cf06f7ec284d.png)